de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/payment/JsonPaymentService.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * #%L
+ * de.metas.business.rest-api-impl
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.rest_api.payment;
+
+import de.metas.banking.api.BankAccountId;
+import de.metas.banking.api.IBPBankAccountDAO;
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.service.IBPartnerDAO;
+import de.metas.money.CurrencyId;
+import de.metas.payment.TenderType;
+import de.metas.payment.api.IPaymentBL;
+import de.metas.rest_api.utils.CurrencyService;
+import de.metas.util.Services;
+import de.metas.util.lang.ExternalId;
+import de.metas.util.time.SystemTime;
+import lombok.NonNull;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.I_C_Payment;
+import org.compiere.util.Env;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Service
+public class JsonPaymentService
+{
+	private final CurrencyService currencyService;
+
+	private final IBPBankAccountDAO bankAccountDAO = Services.get(IBPBankAccountDAO.class);
+	private final IBPartnerDAO partnerDAO = Services.get(IBPartnerDAO.class);
+
+	public JsonPaymentService(final CurrencyService currencyService)
+	{
+		this.currencyService = currencyService;
+	}
+
+	public ResponseEntity<String> createPaymentFromJson(@NonNull @RequestBody final JsonPaymentInfo jsonPaymentInfo)
+	{
+		final IPaymentBL paymentBL = Services.get(IPaymentBL.class);
+		final LocalDate dateTrx = SystemTime.asLocalDate();
```

Comment:
please add this as an optional property to JsonPaymentInfo. only use SystemTime if it's not provided